File: evaluation/src/main/scala/gsd/fms/ImplicationGraph.scala
//package gsd.fms
//
//import dnf.DNFImplBuilder
//import java.io.PrintStream
//
//object ImplicationGraph {
//
//  case class Config(inputFile: String = "",
//                    out: PrintStream = System.out)
//
//  def main(args: Array[String]) {
//    val parser = new scopt.immutable.OptionParser[Config]("ImplicationGraph", "1.0") {
//      def options = Seq(
//        arg("<file>", "Input CXT file") {
//          (file: String, c: Config) => c.copy(inputFile = file)
//        },
//        opt("o", "output", "Output file") {
//          (file: String, c: Config) => c.copy(out = new PrintStream(file))
//        })
//
//    }
//
//    parser.parse(args, Config()) map { c =>
//      val context = CXTParser.parse(c.inputFile)
//      val dnf = context.toDNF
//      val b = new DNFImplBuilder(dnf, context.attributes.size)
//      b.implications foreach println
//    } getOrElse {}
//
//  }
//
//}

File: evaluation/src/main/java/gsd/fms/ImplicationGraph.java
package gsd.fms;

import dk.itu.fms.formula.dnf.DNF;

import java.util.LinkedList;
import java.util.List;

public class ImplicationGraph {

    private DNF dnf;
    private List<Integer[]> implications = null;

    public ImplicationGraph(DNF dnf) {
        this.dnf = dnf;
    }

    public List<Integer[]> calcImplications() {
        List<Integer[]> result = new LinkedList<Integer[]>();
        for (int i : dnf.getVariables()) {
            for (int j : dnf.getVariables()) {
                if (i == j) continue;

                System.out.println("Working on: " + i + "," + j);

                if (dnf.implication(i,j))
                    result.add(new Integer[] {i, j});
            }
        }
        this.implications = result;
        return result;
    }

}
